src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,65 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React, { useState } from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+interface TodoFilterProps {
+  onSelect: (value: string) => void;
+  onChange: (value: string) => void;
+}
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+export const TodoFilter: React.FC<TodoFilterProps> = ({
+  onSelect,
+  onChange,
+}) => {
+  const [value, setValue] = useState('');
+  const handleInput: React.ChangeEventHandler<HTMLInputElement> = event => {
+    onChange(event.target.value);
+    setValue(event.target.value);
```

Comment:
Consider debouncing the `onChange` handler to improve performance, especially if the search operation is expensive or if it triggers a network request.